lgc/patch/llpcPatchDescriptorLoad.cpp

```diff
@@ -536,6 +536,23 @@ Value* PatchDescriptorLoad::LoadBufferDescriptor(
         return UndefValue::get(VectorType::get(builder.getInt32Ty(), 4));
     }
 
+    if ((pNode == pTopNode) && (pNode->type == ResourceNodeType::DescriptorBufferCompact))
+    {
+        // This is a compact buffer descriptor (only two dwords) in the top-level table. We special-case
+        // that to use user data SGPRs directly, if PatchEntryPointMutate managed to fit the value into
+        // user data SGPRs.
+        uint32_t resNodeIdx = pTopNode - m_pPipelineState->GetUserDataNodes().data();
+        auto pIntfData = m_pPipelineState->GetShaderInterfaceData(m_shaderStage);
+        uint32_t argIdx = pIntfData->entryArgIdxs.resNodeValues[resNodeIdx];
+        if (argIdx > 0)
+        {
+            // Resource node isn't spilled. Load its value from function argument.
+            Argument* pDescArg = m_pEntryPoint->getArg(argIdx);
```

Comment:
Maybe call it "pDesc". Traditionally, we don't use "Arg" postfix for those extracted from entry-point unless we need to differentiate which belongs to callees and which belongs to callers. Just a nit.

Reply:
I think pDescArg is fine, to show that it is the descriptor passed as an arg to the function, and in the next line we rely on that to set the IR name of the arg.